README.md

```diff
@@ -1,18 +1,83 @@
 
-## TODOs
-- [X] Set up moogBundle
-- [X] Switch to SAML for authentication.
-- [ ] Use customizing areas and widgets code similar to apos-personas.
-- [ ] Create widgets module to allow client-side Salesforce login and data retrieval (jsForce or SAML)
+Pages and pieces are set to match experiences in their settings modals. Widgets are set to match experiences with a drop-down multi-select field in their contextual UI.
 
+**Note:** This module does *not* enabled logging the user into the Apostrophe application as an Apostrophe user. It sets information about them in their active session, which is used to customize their Apostrophe experience. If you are looking to log the user into the Apostrophe site with a Salesforce SAML SSO, you should probably look at [apostrophe-saml](https://github.com/apostrophecms/apostrophe-saml).
 
-## Options in `data/local.js`:
+## Usage
+- Requires Node 8+ due to use of async/await.
 
-## README
+In addition to project-level configuration of SOQL queries and credentials, there are additional requirements for the SAML connection. Salesforce has specific configuration requirements to set up SSO access. [See their documentation section for all the steps for this.](https://developer.salesforce.com/docs/atlas.en-us.externalidentityImplGuide.meta/externalidentityImplGuide/external_identity_provide_sso_for_web_apps.htm)
 
-1. Get Salesforce identity provider XML file, adding it to connect widgts directory as `salesforce-metadata.xml`.
-2. Create TLS certificate as `connect.cer` and `connect.key`.
\ No newline at end of file
+Once you have the metadata XML file from the connected app, download that, rename it as `salesforce-metadata.xml`, and place it in the `lib/modules/apostrophe-salesforce-connect` directory.
+
+You will also need to create a TLS certificate for the SAML connection. This can be a self-signed certificate.  These are used for SAML assertions, they are not involved in HTTPS. Usually a self-signed certificate will do. The files must be in the `apostrophe-salesforce-connect` directory and be named `connect.cer` and `connect.key`.
+
+From the project root, you can use the following openssl command:
+
+```
+openssl req -new -x509 -days 365 -nodes -sha256 -out lib/modules/apostrophe-salesforce-connect/connect.cer -keyout lib/modules/apostrophe-salesforce-connect/connect.key -days 3650
+```
+
+### Project configuration
+
+The bundle module requires several project-level configurations and also allows for additional optional project-level configurations to enable additional features. As with all module configuration, this is best done in the `data/local.js` file as shown below to avoid committing sensitive information to version control. The queries and other non-sensitive information could instead be configured in the `app.js` module declaration.
```

Comment:
Not all module configuration belongs in data/local.js so the wording here is confusing. And private repos aren't a bad place for confidential stuff necessarily. How about just:

"A private repo can be a reasonable place for these settings, but if you do not wish some of these settings to be present in version control or you want them to vary from server to server, you can use the `data/local.js` file as described here:

https://docs.apostrophecms.org/apostrophe/tutorials/intermediate/deployment

You can also use environment variables, via `process.env.YOUR_ENV_VAR_NAME`."